src/seamless_communication/segment/silero_vad.py

```diff
@@ -0,0 +1,249 @@
+            repo_or_dir="snakers4/silero-vad",
+            model="silero_vad",
+            force_reload=False,
+            onnx=False,
+        )
+        self.sample_rate = sample_rate
+        self.chunk_size_sec = chunk_size_sec
+        self.pause_length = pause_length
+
+    def segment_long_input(self, audio: torch.Tensor) -> None:
+        """
+        Split long input into chunks using speech timestamps.
+        """
+        max_segment_length_samples = self.chunk_size_sec * self.sample_rate
+        pause_length_samples = self.pause_length * self.sample_rate
+
+        speech_timestamps = self.get_speech_timestamps(
+            audio, self.model, sampling_rate=self.sample_rate
+        )
+
+        segments = []
+        current_segment = []
+
+        for segment in speech_timestamps:
+            start_samples = segment[0]
+            end_samples = segment[1]
+
+            if current_segment and (
+                end_samples - current_segment[0] > max_segment_length_samples
+                or start_samples - current_segment[1] > pause_length_samples
+            ):
+                segments.append(current_segment)
+                current_segment = []
+
+            if not current_segment:
+                current_segment = [start_samples, end_samples]
+            else:
+                current_segment[1] = end_samples
+        if current_segment:
+            segments.append(current_segment) 
+
+        return segments
+    
+    def get_speech_timestamps(
+        self,
+        audio: torch.Tensor,
+        model,
+        sampling_rate: int = SAMPLING_RATE,
+        min_speech_duration_ms: int = 500,
+        window_size_samples: int = 1536,
+    ) -> tp.List[tp.Tuple[int, int]]:
+        """
+        Get speech timestamps based on the speech probabilities.
+        """
+        probs, _ = self.get_speech_probs(
+            audio=audio,
+            model=model,
+            sampling_rate=sampling_rate,
+            window_size_samples=window_size_samples,
+        )
+
+        max_segment_length_samples = self.chunk_size_sec * self.sample_rate
+        min_segment_length_samples = min_speech_duration_ms / 1000 * sampling_rate
+
+        segments = self.pdac(
+            probs=probs,
+            max_segment_length=max_segment_length_samples,
+            min_segment_length=min_segment_length_samples,
+            window_size_samples=window_size_samples,
+        )
+
+        speech_timestamps = [(seg.start, seg.end) for seg in segments]
+
+        return speech_timestamps
+
+    def pdac(
+            self,
+            probs: np.array, 
+            max_segment_length: float, 
+            min_segment_length: float, 
+            window_size_samples: float,
+        ) -> tp.List[Segment]:
+        """
+        Recursively splits segments based on speech threshold and duration. 
+        """
+        segments = []
+        sgm = Segment(0, len(probs)*window_size_samples, probs)
+
+        def recursive_split(sgm):
```

Comment:
can you move it into a classmethod or a separate function and then add a unit test for it?